src/DriverTransferUtils.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.20;
+
+import {Drips, StreamReceiver, IERC20, SafeERC20} from "./Drips.sol";
+import {ERC2771Context} from "openzeppelin-contracts/metatx/ERC2771Context.sol";
+
+/// @notice Token transfer utilities for drivers.
+/// Encapsulates the logic for token transfers made by drivers implementing user identities.
+/// Compatible with `Caller` by supporting ERC-2771.
+abstract contract DriverTransferUtils is ERC2771Context {
+    /// @notice The Drips address used by this driver.
+    // slither-disable-next-line naming-convention
+    Drips internal immutable _drips;
```

Comment:
It doesn't matter how many `immutable`s we define, all that matters is how many times it's used. An `immutable` works as a template, wherever it's used in the bytecode, the init code has a gap and the constructor fills it with an actual value. Having 2 immutables is just having 2 ways to fill up the holes, it doesn't cost anything extra.

It's not an elegant solution though to have the same value in 2 places. I didn't want `DriverTransferUtils` to affect the drivers' public APIs (except ERC-2771 context), so I wanted to avoid `Drips public immutable drips;`. A virtual function is a clever workaround though. I've pushed this approach, WDYT? It may fail on Slither, if we like it, we can fix that.